test/blackbox-tests/test-cases/pkg/opam-var/opam-var-pkg.t

```diff
@@ -0,0 +1,227 @@
+  >   [ "echo" "15"   _:bin ]
+  >   [ "echo" "16" foo:bin ]
+  >   [ "echo" "17"   _:sbin ]
+  >   [ "echo" "18" foo:sbin ]
+  >   [ "echo" "19"   _:lib ]
+  >   [ "echo" "20" foo:lib ]
+  >   [ "echo" "21"   _:man ]
+  >   [ "echo" "22" foo:man ]
+  >   [ "echo" "23"   _:doc ]
+  >   [ "echo" "24" foo:doc ]
+  >   [ "echo" "25"   _:share ]
+  >   [ "echo" "26" foo:share ]
+  >   [ "echo" "27"   _:etc ]
+  >   [ "echo" "28" foo:etc ]
+  >   [ "echo" "29"   _:build ]
+  >   [ "echo" "30" foo:build ]
+  >   [ "echo" "31"   _:dev ]
+  >   [ "echo" "32" foo:dev ]
+  >   [ "echo" "33"   _:opamfile ]
+  >   [ "echo" "34" foo:opamfile ]
+  >   [ "echo" "35"     with-test ]
+  >   [ "echo" "36"   _:with-test ]
+  >   [ "echo" "37" foo:with-test ]
+  >   [ "echo" "38"     with-doc ]
+  >   [ "echo" "39"   _:with-doc ]
+  >   [ "echo" "40" foo:with-doc ]
+  >   [ "echo" "41"   _:with-dev-setup ]
+  >   [ "echo" "42" foo:with-dev-setup ]
+  > ]
+  > EOF
+  > mkpkg "foo" <<EOF
+  > opam-version: "2.0"
+  > EOF
+  > solve testpkg
+  Solution for dune.lock:
+  foo.0.0.1
+  testpkg.0.0.1
+  
+Inspecting the lockfile we can see how each opam package variable was translated into a
+corresponding Dune version.
+
+  $ cat dune.lock/testpkg.pkg
+  (version 0.0.1)
+  
+  (build
+   (progn
+    (run echo 1 %{pkg-self:name})
+    (run echo 2 %{pkg-self:name})
+    (run echo 3 %{pkg:name:foo})
+    (run echo 4 %{pkg-self:version})
+    (run echo 5 %{pkg-self:version})
+    (run echo 6 %{pkg:version:foo})
+    (run echo 7 %{pkg-self:depends})
+    (run echo 8 %{pkg:depends:foo})
+    (run echo 9 %{pkg-self:installed})
+    (run echo 10 %{pkg:installed:foo})
+    (run echo 11 %{pkg-self:enable})
+    (run echo 12 %{pkg:enable:foo})
+    (run echo 13 %{pkg-self:pinned})
+    (run echo 14 %{pkg:pinned:foo})
+    (run echo 15 %{pkg-self:bin})
+    (run echo 16 %{pkg:bin:foo})
+    (run echo 17 %{pkg-self:sbin})
+    (run echo 18 %{pkg:sbin:foo})
+    (run echo 19 %{pkg-self:lib})
+    (run echo 20 %{pkg:lib:foo})
+    (run echo 21 %{pkg-self:man})
+    (run echo 22 %{pkg:man:foo})
+    (run echo 23 %{pkg-self:doc})
+    (run echo 24 %{pkg:doc:foo})
+    (run echo 25 %{pkg-self:share})
+    (run echo 26 %{pkg:share:foo})
+    (run echo 27 %{pkg-self:etc})
+    (run echo 28 %{pkg:etc:foo})
+    (run echo 29 %{pkg-self:build})
+    (run echo 30 %{pkg:build:foo})
+    (run echo 31 %{pkg-self:dev})
+    (run echo 32 %{pkg:dev:foo})
+    (run echo 33 %{pkg-self:opamfile})
+    (run echo 34 %{pkg:opamfile:foo})
+    (run echo 35 %{pkg-self:with-test})
+    (run echo 36 %{pkg-self:with-test})
+    (run echo 37 %{pkg:with-test:foo})
+    (run echo 38 %{pkg-self:with-doc})
+    (run echo 39 %{pkg-self:with-doc})
+    (run echo 40 %{pkg:with-doc:foo})
+    (run echo 41 %{pkg-self:with-dev-setup})
+    (run echo 42 %{pkg:with-dev-setup:foo})))
+  
+  (deps foo)
+
+The values here are not important, but Dune should be able to interpret the variables.
+
+  $ build_pkg testpkg
```

Comment:
This will be fixed after this PR. The order of the macro is wrong.

Reply:
Could you either:

1. filter out the backtrace
2. push the fix as a separate commit so there's no error.

Otherwise, this will make every other PR that touches pkg_rules break.